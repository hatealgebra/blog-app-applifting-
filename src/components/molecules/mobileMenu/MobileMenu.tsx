import React from "react";
import MobileMenuContainer, {
  MobileMenuCloseButton,
  MobileMenuNav,
} from "./mobileMenu.styled";
import { GrClose } from "@react-icons/all-files/gr/GrClose";

import { navLinks } from "../../../utils/contants";
import StyledLink, { MobileMenuLink } from "../../atoms/links/link.styled";
import LoginLink from "../../atoms/links/LoginLink";

const MobileMenu = ({ isOpen, setClose }: MobileMenuProps) => {
  return (
    <MobileMenuContainer
      isOpen={isOpen}
      className="mobile-menu"
      data-testid="mobileMenu"
    >
      <div className="mobile-menu__help-div">
        <MobileMenuCloseButton onClick={setClose}>
          <GrClose size="20px" width={10} />
        </MobileMenuCloseButton>
        <MobileMenuNav>
          <MobileMenuLink to={navLinks.MY_ARTICLES}>
            Recent articles
          </MobileMenuLink>
          <MobileMenuLink to={navLinks.MY_ARTICLES}>About</MobileMenuLink>
        </MobileMenuNav>
        <LoginLink />
      </div>
    </MobileMenuContainer>
  );
};

export interface MobileMenuProps {
  isOpen: boolean;
  setClose: () => void;
}

export default MobileMenu;
