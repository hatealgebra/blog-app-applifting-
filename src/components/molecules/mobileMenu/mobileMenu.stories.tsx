import { action } from "@storybook/addon-actions";
import { ComponentMeta } from "@storybook/react";
import React from "react";
import MobileMenu from "./MobileMenu";

export const MobileMenuExample = () => (
  <MobileMenu isOpen={true} setClose={() => action("Mobile menu closed")} />
);

export default {
  title: "Molecules/Mobile Menu",
  component: MobileMenu,
} as ComponentMeta<typeof MobileMenu>;
