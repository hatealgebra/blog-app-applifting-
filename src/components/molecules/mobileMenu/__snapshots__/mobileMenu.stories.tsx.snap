// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/Mobile Menu MobileMenuExample smoke-test 1`] = `
<div class="sc-cwKisF cUdiOH mobile-menu"
     data-testid="mobileMenu"
>
  <div class="mobile-menu__help-div">
    <button class="sc-iJfdHH fSmIGi">
      <svg stroke="currentColor"
           fill="currentColor"
           stroke-width="0"
           viewbox="0 0 24 24"
           width="20px"
           height="20px"
           xmlns="http://www.w3.org/2000/svg"
      >
        <path fill="none"
              stroke="#000"
              stroke-width="2"
              d="M3,3 L21,21 M3,21 L21,3"
        >
        </path>
      </svg>
    </button>
    <nav class="sc-lltjXc hQtBUA">
      <a class="sc-eLSjS sc-bqOBKd cwmAMq bvjZlb">
        Recent articles
      </a>
      <a class="sc-eLSjS sc-bqOBKd cwmAMq bvjZlb">
        About
      </a>
    </nav>
    <a class="sc-ksJisA jEBuxO login-link"
       href="/login"
    >
      <span>
        Login
      </span>
      <svg stroke="currentColor"
           fill="none"
           stroke-width="0"
           viewbox="0 0 24 24"
           height="1em"
           width="1em"
           xmlns="http://www.w3.org/2000/svg"
           style="transform: rotate(180deg);"
      >
        <path stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              d="M10 19l-7-7m0 0l7-7m-7 7h18"
        >
        </path>
      </svg>
    </a>
  </div>
</div>
`;
